Automattic/jetpack
# Adds modules enforcing logic.

This change introduces a new method to enforce module loading with an additional side effect of the `jetpack_pre_module_activate` being run on first activation.

#### Changes proposed in this Pull Request:

* Added the `enforce` method to the `Modules` class.
* Added unit tests.

#### Other information:

#### Jetpack product discussion
TBA

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:






* Install a standalone plugin that depends on the Jetpack module being activated.
* Set up a filter to run on the module activation action.
```php 
add_action( 'jetpack_pre_activate_module', 'l' ); //Use any other handler here instead of l if you wish
```
* Use the new method to enforce the needed module to load.
* Make sure the module activation hook fires only once.

##### Additional testing using a specific use case by VideoPress
D86939-code
